gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/BuildController.groovy

```diff
@@ -41,8 +45,12 @@ class BuildController {
 
   @ApiOperation(value = "Get build masters", response = List.class)
   @RequestMapping(value = "v2/builds", method = RequestMethod.GET)
-  List<String> getBuildMasters() {
-    buildService.getBuildMasters()
+  List<String> getBuildMasters(@RequestParam(value = "type", defaultValue = "") String type) {
+    if (type == "") {
+      buildService.getBuildMasters()
```

Comment:
Just a minor point, but it might make make sense to just call `getBuildMasters(type)` directly here, and then have the `if` statement in `getBuildMasters` to decide whether to call the API with/without a type.  (Might avoid some of the duplication between `getBuildMasters` and `getBuildMastersOfType`, though they're admittedly pretty short functions.) . Either way, this is fine too.

Reply:
updated